src/vivarium_gates_mncnh/utilities.py

```diff
@@ -181,3 +182,16 @@ def get_random_variable(draw: int, seeded_distribution: SeededDistribution) -> f
 
 def get_location(builder: Builder) -> str:
     return builder.data.load("population.location")
+
+
+def rate_to_probability(
+    rate: Sequence[float] | NumberLike, duration: int | None = None
+) -> NumericArray:
+    # encountered underflow from rate > 30k
+    # for rates greater than 250, exp(-rate) evaluates to 1e-109
+    # beware machine-specific floating point issues
+
+    rate = np.array(rate)
+    rate[rate > 250] = 250.0
+    probability: NumericArray = 1 - np.exp(-rate)
+    return probability
```

Comment:
You forgot to do the scaling. This can also be more simply done like this:
```python
from vivarium.framework.utilities import rate_to_probability as rate_to_probability_

def rate_to_probability(
    rate: Sequence[float] | NumberLike, scaling_factor: float = 1.0
) -> NumericArray:
    return rate_to_probability_(rate * scaling_factor)
```